currency_rate_update/models/currency_rate_update.py

```diff
@@ -141,7 +141,7 @@ def refresh_currency(self):
                 # No need to test if main_currency exists, because it is a
                 # required field
                 if float_compare(
-                        main_currency.rate, 1,
+                        main_currency.rate.with_context(company_id=company.id), 1,
```

Comment:
I think it should be `main_currency.with_context(company_id=company.id).rate` instead